src/DynamoSandbox/Program.cs

```diff
@@ -237,13 +241,31 @@ public static void Main(string[] args)
             try
             {
                 var cmdLineArgs = CommandLineArguments.FromArguments(args);
+                string locale;
 
                 if (!string.IsNullOrEmpty(cmdLineArgs.Locale))
                 {
                     // Change the application locale, if a locale information is supplied.
                     Thread.CurrentThread.CurrentUICulture = new CultureInfo(cmdLineArgs.Locale);
                     Thread.CurrentThread.CurrentCulture = new CultureInfo(cmdLineArgs.Locale);
+                    locale = cmdLineArgs.Locale;
+                }
+                else
+                {
+                    var supportedLocale = new HashSet<string>(new[]
+                        {
+                            "cs-CZ", "de-DE", "en-US", "es-ES", "fr-FR", "it-IT",
+                            "ja-JP", "ko-KR", "pl-PL", "pt-BR", "ru-RU", "zh-CN", "zh-TW"
+                        });
+                    // in case no language is specified, the locale of libG should be that of the operating
+                    // system. If that locale is not supported by Dynamo, default to en-US.
+                    var systemLocale = CultureInfo.InstalledUICulture.ToString();
+                    locale = supportedLocale.Contains(systemLocale) ? systemLocale : "en-US";
```

Comment:
Oops, my comment about making this case-insensitive comparison is gone, can we try this?

```csharp
supportedLocale.Any(x => x.Compare(systemLocal, BlaBla.CaseInsensitive))
```

Do try it out for real though, my Linq is rusty.